src/sql/src/plan/statement/ddl.rs

```diff
@@ -3542,14 +3543,54 @@ generate_extracted_config!(
     (EnableLetrecFixpointAnalysis, Option<bool>, Default(None))
 );
 
+/// Convert a [`CreateClusterStatement`] into a [`Plan`].
+///
+/// The reverse of [`unplan_create_cluster`].
 pub fn plan_create_cluster(
+    scx: &StatementContext,
+    stmt: CreateClusterStatement<Aug>,
+) -> Result<Plan, PlanError> {
+    let plan = plan_create_cluster_inner(scx, stmt)?;
+
+    // Roundtrip through unplan and make sure that we end up with the same plan.
+    if let CreateClusterVariant::Managed(_) = &plan.variant {
+        let stmt = unplan_create_cluster(scx, plan.clone())
+            .map_err(|e| PlanError::Replan(e.to_string()))?;
+        let create_sql = stmt.to_ast_string_stable();
+        let stmt = parse::parse(&create_sql)
+            .map_err(|e| PlanError::Replan(e.to_string()))?
+            .into_element()
+            .ast;
+        let (stmt, _resolved_ids) =
+            names::resolve(scx.catalog, stmt).map_err(|e| PlanError::Replan(e.to_string()))?;
+        let stmt = match stmt {
+            Statement::CreateCluster(stmt) => stmt,
+            stmt => {
+                return Err(PlanError::Replan(format!(
+                "replan does not match: plan={plan:?}, create_sql={create_sql:?}, stmt={stmt:?}"
+            )))
+            }
+        };
+        let replan =
+            plan_create_cluster_inner(scx, stmt).map_err(|e| PlanError::Replan(e.to_string()))?;
+        if plan != replan {
```

Comment:
This should also be in `ALTER CLUSTER`.

Reply:
What comparison should we do for `ALTER CLUSTER`? We don't have a `CreateClusterPlan` to compare to.

Reply:
For that I think we'd have to move the check into sequencing and do the full roundtrip of `Cluster -> CreateClusterPlan -> CreateClusterStatement -> String -> CreateClusterStatement -> CreateClusterPlan -> Cluster`.

Reply:
Hmm yeah. ALTER has an updated catalog so it could produce a sql string from that to plan. But we need to compare it to the `Cluster` object in the catalog, which is the sequenced plan, not the plan? I'm ok merging as is, but I do very much think we should complete the thought because `ALTER` seems like a likely place where we make a small mistake and either corrupt or have silent mutation to some object.

Reply:
It sounds like this commit doesn't add any new hazards to `ALTER CLUSTER`, but this feature could be used as a tool to protect against existing or future hazards in `ALTER CLUSTER`? If I'm understanding that correctly, then I'm inclined to merge as is and let someone else add that to `ALTER CLUSTER` as a follow up.